ops/charm.py

```diff
@@ -901,9 +905,10 @@ class SecretRemoveEvent(SecretEvent):
     observers have updated to that new revision, this event will be fired to
     inform the secret owner that the old revision can be removed.
 
-    Typically, the charm will call
+    After any required cleanup, the charm should call
     :meth:`event.secret.remove_revision() <ops.Secret.remove_revision>` to
-    remove the now-unused revision.
+    remove the now-unused revision. If the charm does not, then the event will
+    be emitted again, when further revisions are ready for removal.
```

Comment:
Does that result in N^2 event pile-up if the charm is stubborn?

Reply:
If the charm never actually does the remove then they'll get *n* events every time a new revision is obsolete, where *n* is the number of obsolete revisions. So for the secret owner it's more of an N pile-up than N^2 I would say. Still not great, especially if you move to new revisions frequently.

Reply:
I guess I meant N^2 total runtime complexity, got it.

Anyway, that's not an issue with the PR, rather with the general secret revision protocol.